### PR TITLE
Fix AttributeError: 'NoneType' object has no attribute 'name'

### PR DESCRIPTION
**Pull request by drewgillson**
There is a bug related to the interplay between the society of mind agent and the FileLogger, probably somewhere around https://github.com/microsoft/autogen/blob/main/autogen/agentchat/contrib/society_of_mind_agent.py#L181, which causes an empty name property to be sent. The FileLogger should gracefully handle this circumstance and not return an error to the function caller.

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: autogen/logger/file_logger.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def log_chat_completion(
         thread_id = threading.get_ident()
         source_name = None
         if isinstance(source, str):
-            source_name = source
+            source_name = getattr(source, "name", "unknown")
         else:
             source_name = source.name
```
Comment (sonichi) [inline autogen/logger/file_logger.py:94]:
```suggestion
            source_name = source
        else:
            source_name = getattr(source, "name", "unknown")
```

**Review by sonichi — COMMENTED** [review]
There is an error in the change. It'll be good to write a simple test to cover the case.

File: autogen/logger/file_logger.py (hunk 1 above)
Comment (Hk669) [reply autogen/logger/file_logger.py:94]: this looks good. @drewgillson, can you please update this and add tests as mentioned.

**Comment by Hk669** [comment]
@drewgillson thanks for spotting the bug.

**Review by jackgerrits — APPROVED** [review]

**Pull request merged**